Nitzan/eco 553/multiple users devices wallets

nitzantomer opened the pull request.

nitzantomer committed:
Nitzan/eco 754/multiple wallets user model change (#230)

* add duration to rate limit prefix (#228)

* add duration to rate limit prefix

This way when there are 2 rate limits with different durations but the same prefix they don't collide
Also add rate limit on wallet earn, so a user can't earn too much into a single wallet

* first check the rate and then increment event

* style

* pr fixes

* fix unhandeled promise rejection in tests

* fix tests

* show offer content in admin (#231)

* changing the user model and added the wallets model

* unit tests now pass

* fixes for the system test

* rebase from master

* PR fixes

* fixed PR comments

nitzantomer committed:
Nitzan/eco 753/update register (#233)

* changing the user model and added the wallets model

* unit tests now pass

* changing the user model and added the wallets model

* fixes for the system test

* fixed PR comments

* auth token response now return whether the user is new or returning

* rebase

* metrics and no wallet error
fixed the metrics a bit
checking for wallet existence before creating orders

* fixed admin js location change

* Nitzan/eco 752/logout (#235)

* changing the user model and added the wallets model

* unit tests now pass

* added the logout endpoint/functionality

* PR fixes

* order history only returns order for current wallet (#236)

for that i also changed the orders contexts table/model to include the wallet.

* fix: make sure wallet exists

nitzantomer committed:
v2: support old versions clients

doodyparizada commented on scripts/src/public/services/users.ts:
```diff
@@ -124,15 +123,78 @@ export type UserStats = {
+	};
+}
+
+async function createUserProfile(user: User): Promise<UserProfile> {
```
weird name for something that queries the DB and doesn't create anything

doodyparizada reviewed (commented):
1. tests
2. instead of Old use V1 prefix
3. If you can move all the V1 stuff to a separate file it would be nice

nitzantomer committed:
fixed PR comments

nitzantomer committed:
unit tests passing
including tests for v1

nitzantomer committed:
system tests pass for v2

nitzantomer committed:
user_id and device_id mandatory in all JWTS
including system tests for v2

nitzantomer committed:
updated jwt-service image

nitzantomer committed:
once again, updated the jwt-service image

nitzantomer committed:
jwt valiation and system test

nitzantomer committed:
v1 and v2 compatible
including unit and system tests for both versions
lacks tests for the new multi scenarios

ravitb commented on scripts/src/admin/services.ts:
```diff
@@ -337,20 +336,24 @@ async function orderToHtml(order: Order): Promise<string> {
 async function userToHtml(user: User): Promise<string> {
+	const accounts = (await user.getWallets()).all().map(wallet => {
+		return `
+		<a href="${ BLOCKCHAIN.horizon_url}/accounts/${ wallet.address }">${ wallet.address }</a>
```
You could use DomParser, much more readable for html chunk of code.

nitzantomer replied:
the entire admin is written in this awful way, we need to change all of it and go with react

doodyparizada commented on scripts/src/client.ts:
```diff
@@ -61,13 +61,13 @@ export class ClientError extends Error {
-	public static async create(data: { device_id: string; }, headers?: StringMap) {
-		const res = await axios.post<AuthToken>(MARKETPLACE_BASE + "/v1/users", data, { headers });
-		return new ClientRequests(res.data);
+	public static async create(data: any, headers?: StringMap) {
```
why `any`?

doodyparizada reviewed (commented):
1. Would like to see less use of `any`
2. Would like to reduce duplication of code by implementing v1 as a lightweight layer over v2.

nitzantomer committed:
seperated file for v1 order services

nitzantomer committed:
use case: two users with the same wallet
including system test for it

nitzantomer committed:
added tests for multiple users, wallets and devices
the tests include unit tests and the system tests.
fixes were made which the tests found

nitzantomer committed:
fixed unit tests

nitzantomer committed:
after rebase from master

nitzantomer committed:
changed the orders unit tests a bit

nitzantomer committed:
rebase from master and doody's logoutfix

nitzantomer committed:
added logs for failed tests in travis

doodyparizada committed:
fix p2p error

doodyparizada committed:
fix test

doodyparizada committed:
remove console logs

doodyparizada committed:
fix

doodyparizada commented on scripts/src/public/services/orders.ts:
```diff
@@ -398,23 +461,38 @@ function openOrderDbToApi(order: db.Order, userId: string): OpenOrder {
+		}, pick(context.meta, "title", "description", "content", "call_to_action"));
+	} else {
+		context = order.contextForWallet(wallet)!;
+		const app = (await Application.findOneById(context.user.appId))!;  // TODO This is run inside a for loop - better remove this
```
This is run inside a for loop - better remove this

nitzantomer committed:
fixed the changed title/description for orders (#261)

* fixed the changed title/description for orders

* fixed to context.type instead of order.type

nitzantomer committed:
added current wallet to register response (#262)

nitzantomer committed:
cache apps list

nitzantomer committed:
added db index and removed comments
the index is for orders_contexts.wallet

nitzantomer committed:
added bi events

nitzantomer committed:
merge with master

nitzantomer committed:
afte merge fixes

doodyparizada reviewed (approved):
Make this changes in a separate PR
